File: 121.best-time-to-buy-and-sell-stock.cpp
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int buy = INT_MAX, sell = 0;
        for (int i = 0; i < prices.size(); i++)
        {
            buy = min(buy, prices[i]);
            sell = max(prices[i] - buy, sell);
        }
        return sell;
    }
};

File: 121.best-time-to-buy-and-sell-stock.py
#
# @lc app=leetcode id=121 lang=python3
#
# [121] Best Time to Buy and Sell Stock
#

# @lc code=start
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]
        profit = 0

        for num in prices:
            if num < buy:
                buy = num
                continue
            diff = num - buy
            if diff > profit:
                profit = diff

        return profit
# @lc code=end

